### ISSUE TITLE
code line cannot actually matches the line number

### ISSUE THREAD

**Issue by dreamer2q** [baderouaich/flutter_syntax_view#8]
From https://widget.studio, when you change the font size, there may occur that actually code line cannot match line number indicator.

see screenshot for more info

![image](https://user-images.githubusercontent.com/56211747/111333232-75dfd680-86ad-11eb-9552-261b3ea5b855.png)

**Comment by BaderEddineOuaich** [comment]
Hi @dreamer2q, i believe the syntax view version used in https://widget.studio is a [fork](https://github.com/rodydavis/flutter_syntax_view) by @rodydavis as he mentioned in issue #6 .
Although we are trying to find a better solution to that, as we are talking i have made some changes in version 4.2.2 which i think solves #6 (still waiting for tests in your side before publishing)
@rodydavis can you give v4.2.2 a test in https://widget.studio to see if this issue solved? 
Thank you!

**Comment by dreamer2q** [comment]
It works much better without softWrap.

But if you enable softWrap and increase the font size, code line will overflow. 

![Screenshot_20210317_011901](https://user-images.githubusercontent.com/56211747/111352101-d1669000-86be-11eb-99cb-0490f873d0d8.jpg)

**Comment by BaderEddineOuaich** [comment]
> It works much better without softWrap.
> 
> But if you enable softWrap and increase the font size, code line will overflow.
> 
> ![Screenshot_20210317_011901](https://user-images.githubusercontent.com/56211747/111352101-d1669000-86be-11eb-99cb-0490f873d0d8.jpg)

can you provide the code u used in this example?

**Comment by BaderEddineOuaich** [comment]
Ah yep you're right, I have tested it again with larger font size and this problem show up...
any suggestions to solve this one guys?

**Comment by maheshmnj** [comment]
You just have to calculate the number of lines ("\n") in the text of the textfield and that will give you the number of lines. You could do this using RegExp.

**Comment by rodydavis** [comment]
I will update and let you know if it is fixed! I can also contribute back a custom TextEditingController that highlights the code and allows you to use this in a text field.

Here is a demo:

https://rodydavis.github.io/flutter_analyzer/

If you are interest in looking at the code:

https://github.com/rodydavis/flutter_analyzer/blob/master/lib/src/controller.dart

**Comment by BaderEddineOuaich** [comment]
> You just have to calculate the number of lines ("\n") in the text of the textfield and that will give you the number of lines. You could do this using RegExp.

yes this will work fine until softWrap is enabled, then RichText will break lines to fit the screen leaving text code unchanged. the lines will be broken in the view but text will remain the same...

**Comment by BaderEddineOuaich** [comment]
> I will update and let you know if it is fixed! I can also contribute back a custom TextEditingController that highlights the code and allows you to use this in a text field.
> 
> Here is a demo:
> 
> https://rodydavis.github.io/flutter_analyzer/
> 
> If you are interest in looking at the code:
> 
> https://github.com/rodydavis/flutter_analyzer/blob/master/lib/src/controller.dart

That would be great 😀!  thank you @rodydavis!

**Comment by baderouaich** [comment]
Fixed in 4.1.2 225cd9c1e8ca943ae9748487750e3a6b3842ea57

**Comment by rodydavis** [comment]
Thank you!!
